plugins/utils/__init__.py

```diff
@@ -1077,6 +1077,135 @@ def execute(self, ctx):
             dataset.name = new_name
 
 
+class CloneDataset(foo.Operator):
+    @property
+    def config(self):
+        return foo.OperatorConfig(
+            name="clone_dataset",
+            label="Clone dataset",
+            light_icon="/assets/icon-light.svg",
+            dark_icon="/assets/icon-dark.svg",
+            dynamic=True,
+        )
+
+    def resolve_input(self, ctx):
+        inputs = types.Object()
+
+        _get_clone_dataset_inputs(ctx, inputs)
+
+        return types.Property(inputs, view=types.View(label="Clone dataset"))
+
+    def execute(self, ctx):
+        name = ctx.params["name"]
+        new_name = ctx.params["new_name"]
```

Comment:
do we need to check if the new name is invalid and throw an error if that is true before executing?

Reply:
Validation of `new_name` is performed in `resolve_input()` to make sure the name isn't already taken.

If someone else created a `new_name` dataset between when `resolve_input()` and `execute()` are executed, then `clone()` will raise an informative error that informs the user of it.